package_nso_to_oc/xe/xe_routing_policy.py

```diff
@@ -0,0 +1,521 @@
+
+        for rule_index, rule in enumerate(access.get("as-path-rule", [])):
+            if not "operation" in rule:
+                continue
+            if rule["operation"] == "deny":
+                all_processed = False
+                routing_policy_notes.append(
+f"""
+AS Path Name: {access.get("name")}
+Rule: {rule.get("rule")}
+This rule contains a deny operation, which is not supported in OpenConfig. Translation to OC was skipped.
+""")
+                continue
+
+            members.append(rule["rule"])
+
+            # Ensure the value we're nullifying does exist
+            if common.get_index_or_default(rule_list_after, rule_index, None):
+                rule_list_after[rule_index] = None
+
+        as_path_sets["openconfig-bgp-policy:as-path-sets"]["openconfig-bgp-policy:as-path-set"].append(new_path_set)
+
+        if all_processed:
+            common.get_index_or_default(access_list_after, access_index, {})["name"] = None
+
+    openconfig_routing_policies["openconfig-routing-policy:routing-policy"]["openconfig-routing-policy:defined-sets"]["openconfig-bgp-policy:bgp-defined-sets"].update(as_path_sets)
+
+def process_community_sets(config_before, config_after):
+    community_sets = {"openconfig-bgp-policy:community-sets": {"openconfig-bgp-policy:community-set": []}}
+    community_list = config_before.get("tailf-ned-cisco-ios:ip", {}).get("community-list", {})
+    community_list_after = config_after.get("tailf-ned-cisco-ios:ip", {}).get("community-list", {})
+    process_community_members(community_sets, "standard", community_list, community_list_after)
+    process_community_members(community_sets, "expanded", community_list, community_list_after)
+    openconfig_routing_policies["openconfig-routing-policy:routing-policy"]["openconfig-routing-policy:defined-sets"]["openconfig-bgp-policy:bgp-defined-sets"].update(community_sets)
+
+def process_community_members(community_sets, type, community_list, community_list_after):
+    all_processed = True
+
+    for community_index, community in enumerate(community_list.get(type, [])):
+        new_community_set = {
+            "openconfig-bgp-policy:community-set-name": community.get("name"),
+            "openconfig-bgp-policy:config": {
+                "openconfig-bgp-policy:community-set-name": community.get("name"),
+                "openconfig-bgp-policy:match-set-options": "ANY", # IOS only supports ANY
+                "openconfig-bgp-policy:community-member": []
+            }
+        }
+        members = new_community_set["openconfig-bgp-policy:config"]["openconfig-bgp-policy:community-member"]
+        entry_after = common.get_index_or_default(community_list_after.get(type, []), community_index, {}).get("entry", [])
+
+        for entry_index, entry in enumerate(community.get("entry", [])):
+            if not "expr" in entry:
+                continue
+            if entry["expr"].startswith("deny"):
```

Comment:
If deny, we must note, keep processing for other notes, but can't appended to OC.